src/utils/ProtectedRoute.jsx

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect } from 'react';
+import { Navigate } from 'react-router-dom';
+import { PropTypes, instanceOf } from 'prop-types';
+import { withCookies, cookieKeys, Cookies, clearCookies } from './cookie_utils';
+import { refreshToken } from './auth_utils';
+
+const userIsAuthenticated = async (roles, cookies) => {
+  try {
+    const accessToken = await refreshToken(cookies);
+    if (!accessToken) {
+      return false;
+    }
+    // const loggedIn = await NPOBackend.get(`/auth/verifyToken/${accessToken}`);
+    // return roles.includes(cookies.get(cookieKeys.ROLE)) && loggedIn.status === 200;
+    return roles.includes(cookies.get(cookieKeys.ROLE));
+  } catch (err) {
+    clearCookies(cookies);
+    return false;
+  }
+};
+
+/**
+ * Protects a route from unauthenticated users
+ * @param {Component} children The component the user is trying to access
+ * @param {string} redirectPath The path to redirect the user to if they're not logged in
+ * @param {Array} roles A list of roles that are allowed to access the route
+ * @param {Cookies} cookies The user's current cookies
+ * @returns The relevant path to redirect the user to depending on authentication state.
+ */
+const ProtectedRoute = ({ Component, redirectPath, roles, cookies }) => {
```

Comment:
⚠️ **[eslint]** <react-refresh/only-export-components> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Fast refresh only works when a file only exports components. Move your component(s) to a separate file.